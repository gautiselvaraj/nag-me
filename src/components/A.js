import React from 'react';
import styled from 'styled-components';

const Link = styled.a`
  color: ${props => props.theme.main};
  text-decoration: none;

  &:hover {
    color: ${props => props.theme.mainLighten};
  }
`;

export default ({children, ...otherProps}) => <Link {...otherProps}>{children}</Link>
