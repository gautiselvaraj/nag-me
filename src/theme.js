export default {
  main:           '#214154',
  mainDarken:     '#083148',
  mainLighten:    '#215a7b',
  black:          '#111111',
  white:          '#f6f6f6',
  greyLightest:   '#efefef',
  greyLighter:    '#dddddd',
  greyLight:      '#a5a5a5',
  grey:           '#898989',
  greyDark:       '#777777',
  greyDarker:     '#595959',
  greyDarkest:    '#222',
}
