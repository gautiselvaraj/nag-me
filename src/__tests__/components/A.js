import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import A from '../../components/A';
import theme from '../../theme';
import 'jest-styled-components';

describe('<A />', () => {
  describe('default link', () => {
    it('should match snapshot', () => {
      const tree = renderer
        .create(
          <A href="https://www.gauti.info/nag-me" theme={theme}>
            Link
          </A>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should have correct text, element & color', () => {
      const link = mount(
        <A href="https://www.gauti.info/nag-me" theme={theme}>
          Link
        </A>
      );
      expect(link.text()).toBe('Link');
      expect(link.find('a').length).toBe(1);
      expect(link).toHaveStyleRule('color', theme.main);
    });
  });

  describe('inverse link', () => {
    it('should match snapshot', () => {
      const tree = renderer
        .create(
          <A inverse href="https://www.gauti.info/nag-me" theme={theme}>
            Link
          </A>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should have correct element & color based on inverse prop', () => {
      const link = mount(
        <A inverse href="https://www.gauti.info/nag-me" theme={theme}>
          Link
        </A>
      );
      expect(link.find('a').length).toBe(1);
      expect(link).toHaveStyleRule('color', theme.white);
    });
  });
});
