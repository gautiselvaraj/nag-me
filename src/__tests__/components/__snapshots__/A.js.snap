// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<A /> default link should match snapshot 1`] = `
.c0 {
  color: #214154;
  text-decoration: none;
}

.c0:hover {
  color: #3B5B6E;
}

<a
  className="c0"
  href="https://www.gauti.info/nag-me"
>
  Link
</a>
`;

exports[`<A /> inverse link should match snapshot 1`] = `
.c0 {
  color: #f6f6f6;
  text-decoration: none;
}

.c0:hover {
  color: #dddddd;
}

<a
  className="c0"
  href="https://www.gauti.info/nag-me"
>
  Link
</a>
`;
